### PR TITLE
Fix deadlock when the User directory doesn't exist and with GPU determinism

### PR DESCRIPTION
**Pull request by flyinghead**
Issue #68 
Issue #32

### CODE REVIEW

**Comment by i30817** [comment]
I was wondering if the second bug exists on upstream. Seems like a awkward fix. Have you thought about asking upstream what's happening?

It appears to be that the fix is turning off event queuing always (setpassthrough never becoming true again). Which means that there is some code elsewhere that tries to push a event into the queue (this queue doesn't seem directly linked to a thread, though it has thread safety) and crashes if it's run with event processing delay.

This looks kind of broken to me.

The only thing that calls 'pushevents' is  VideoBackendBase.cpp, on init and `u32 VideoBackendBase::Video_AccessEFB(EFBAccessType type, u32 x, u32 y, u32 data)` which can push both false and true to the 'blocking' argument. Blocking is 'wait for all the events enqueued to be processed' and m_passthrough (which is handled first) is 'handle each event right away or collect them into the queue first'.

Between these calls there is a call to `Fifo::RunGpu();` which is where the the threading comes in. 

Basically i'm worried that turning off m_passthrough will affect performance and the crash shouldn't really be happening if the misbehaving event was thread-safe.

**Comment by flyinghead** [comment]
passthrough must be false because rendering must be done in the rendering thread, which is the thread calling `retro_run()` and ultimately calling `Fifo::RunGpu()`. The emulator thread runs concurrently but it must not handle rendering events itself.

This doesn't happen upstream because either there's a single thread doing everything, or the gpu thread doesn't stop while the emulator is running.

**Comment by i30817** [comment]
> passthrough must be false because rendering must be done in the rendering thread

This is supposed to occur on the original code where the function that got the ifdefn \_\_LIBRETRO\_\_ in the commit already turned it off at the start and on at the end. So if there is a threading violation in the libretro fork it must be elsewhere. That was my point, such as it was...
